tests/main/apparmor-prompting-integration-tests/task.yaml

```diff
@@ -0,0 +1,110 @@
+    snap install prompting-client
+
+restore: |
+    #shellcheck disable=SC2016
+    tests.session -u test exec sh -c 'rm -rf "$HOME/integration-tests"' # $HOME must be that of test user
+    tests.session restore -u test
+
+debug: |
+    TEST_UID="$(id -u test)"
+    echo "outstanding prompts:"
+    snap debug api "/v2/interfaces/requests/prompts?user-id=$TEST_UID"
+    echo "rules:"
+    snap debug api "/v2/interfaces/requests/rules?user-id=$TEST_UID"
+
+execute: |
+    echo "Enable prompting via snap client where possible"
+    # Prompting is unsupported everywhere but the Ubuntu non-core systems with
+    # kernels which support apparmor prompting
+    if ! os.query is-ubuntu || os.query is-core || ! grep 'prompt' /sys/kernel/security/apparmor/features/policy/permstable32 ; then
+        not snap set system experimental.apparmor-prompting=true >& err.out
+        if os.query is-core ; then
+            MATCH "cannot enable prompting feature as it is not supported on Ubuntu Core systems" < err.out
+        else
+            MATCH "cannot enable prompting feature as it is not supported by the system" < err.out
+        fi
+
+        # even if unsupported setting it to false should succeed
+        snap set system experimental.apparmor-prompting=false
+        exit 0
+    fi
+
+    echo "Enable AppArmor prompting experimental feature"
+    snap set system experimental.apparmor-prompting=true
+
+    # Wait for snapd to begin restart
+    sleep 1
+
+    echo "Wait until prompting is active"
+    retry --wait 1 -n 60 systemctl is-active snapd
+
+    echo "Check that shell script and scripted replies exist for $VARIANT"
+    test -f "$VARIANT.sh"
+    test -f "$VARIANT.json"
+
+    # Create unique tmpdir and copy the script to it.
+    # The tmpdir must be in the test user's home directory.
+    #shellcheck disable=SC2016
+    TEST_DIR="$(tests.session -u test exec sh -c 'mktemp --directory --tmpdir="$HOME/integration-tests"')"
+    cp "$VARIANT.sh" "${TEST_DIR}/${VARIANT}.sh"
+    chown test "${TEST_DIR}/${VARIANT}.sh"
+    cp "$VARIANT.json" "${TEST_DIR}/script.json"
+    chown test "${TEST_DIR}/script.json"
+
+    # Test scripts can rely on the scripted client writing their result to
+    # "${TEST_DIR}/result". Those results will be "success" if the scripted
+    # client exits without error, or an error message if it encounters an
+    # unexpected prompt or an error from snapd.
+
+    echo "Run the prompting client in scripted mode in the background as the test user"
+    tests.session -u test exec prompting-client.scripted \
+        --script="${TEST_DIR}/script.json" \
+        --grace-period=1 \
+        --var="BASE_PATH:${TEST_DIR}" | tee "${TEST_DIR}/result" &
+
+    timeout 5 sh -xc 'while ! pgrep -f "prompting-client-scripted" > /dev/null; do sleep 0.1; done'
+    sleep 1 # wait for scripted client to start listening, otherwise it may miss the first prompt
```

Comment:
this sleep seems fragile? can't we use something like lsof or something to see if the connection is there?